### PR TITLE
fix problem with zero EEst and PIDController

### PR DESCRIPTION
**Pull request by ranocha**
I noticed some trouble with `PIDController`s when the error estimates vanish. This fixes the problem.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -38,6 +38,7 @@ if GROUP == "All" || GROUP == "InterfaceI" || GROUP == "Interface"
   @time @safetestset "Composite Interpolation Tests" begin include("interface/composite_interpolation.jl") end
   @time @safetestset "Export tests" begin include("interface/export_tests.jl") end
   @time @safetestset "Type Handling Tests" begin include("interface/type_handling.jl") end
+  @time @safetestset "Controller Tests" begin include("interface/controllers.jl") end
```
Comment (ChrisRackauckas) [inline test/runtests.jl:41]: oh no, was that missing before? Does this have a test case for this?

Comment (ranocha) [reply test/runtests.jl:41]: I added this test case here (but noticed that I forgot to push it in the first commit - fixed in e886266e). This file contains a new test case for this specific case.

Comment (ChrisRackauckas) [reply test/runtests.jl:41]: Ahh okay. Slap a few other controllers in there and see if they have a similar issue with 0 EEst?

**Review by ChrisRackauckas — APPROVED** [review]

File: test/runtests.jl (hunk 1 above)
Comment (ranocha) [reply test/runtests.jl:41]: I did that - everything looks good so far

**Pull request merged**